### PR TITLE
Helm chart 3 upgrade changes

### PR DESCRIPTION
**Pull request by prashant-khunte**
### Change description ###
Helm chart 3 upgrade changes
Spring boot upgrade 2.3.0

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -235,7 +237,7 @@ dependencies {
   compile group: 'org.springframework.cloud', name: 'spring-cloud-starter-openfeign', version: '2.2.1.RELEASE'
   compile group: 'io.github.openfeign', name: 'feign-jackson', version: '10.1.0'
   compile group: 'org.jboss.aerogear', name: 'aerogear-otp-java', version:'1.0.0'
-  compile group: 'uk.gov.hmcts.reform', name: 'health-spring-boot-starter', version: '0.0.3'
+  //compile group: 'uk.gov.hmcts.reform', name: 'health-spring-boot-starter', version: '0.0.3'
```
Comment (amitchidrewar1301) [inline build.gradle:240]: Please remove the commented line

File: src/main/resources/application-local.yaml [hunk 2]
```diff
@@ -9,9 +9,9 @@ spring:
     name: bulk-scanning
   datasource:
     driver-class-name: org.postgresql.Driver
-    url: jdbc:postgresql://localhost:5432/bulk-scan
-    username: postgres
-    password: admin
+    url: jdbc:postgresql://localhost:5432/bspayment
```
Comment (amitchidrewar1301) [inline src/main/resources/application-local.yaml:12]: please revert application local changes

File: charts/ccpay-bulkscanning-api/Chart.yaml [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
 description: A Helm chart for ccpay bulkscanning payments api
 name: ccpay-bulkscanning-api
 home: https://github.com/hmcts/ccpay-bulkscanning-app
-version: 1.0.6
+version: 1.0.8
 maintainers:
   - name: HMCTS Fees and Pay team
+dependencies:
+  - name: java
+    version: ~3.0.0-beta
```
Comment (timja) [inline charts/ccpay-bulkscanning-api/Chart.yaml:11]: do not use a beta version when there's newer GA versions:
https://github.com/hmcts/chart-java/releases

File: src/main/resources/application-local.yaml [hunk 4]
```diff
@@ -36,7 +40,7 @@ spring:
 trusted:
   s2s:
     service:
-      names: ccpay_bubble,cmc,bulk_scan_payment_processor,api_gw,probate_frontend,divorce_frontend
+      names: ccpay_bubble,cmc,bulk_scan_payment_processor,api_gw,probate_frontend,divorce_frontend,payment_app
```
Comment (timja) [inline src/main/resources/application-local.yaml:43]: change looks unrelated

**Review by timja — CHANGES_REQUESTED** [review]
use a GA chart version

**Review by timja — APPROVED** [review]

File: charts/ccpay-bulkscanning-api/Chart.yaml (hunk 3 above)
Comment (amitchidrewar1301) [reply charts/ccpay-bulkscanning-api/Chart.yaml]: @timja  I have updated the chart version to 3.4.4 which is the latest one as per repo. Please confirm if it should be ok to use.

File: config/owasp/suppressions.xml [hunk 5]
```diff
@@ -42,8 +42,8 @@
   </suppress>
 
   <suppress>
-    <notes>This is transitive dependency coming for spring boot, CVE is for tomcat-embed-core-9.0.31.jar and tomcat-embed-websocket-9.0.31.jar</notes>
-    <cve>CVE-2020-13943</cve>
+    <notes>This is transitive dependency coming for spring boot starter, CVE is for tomcat-embed-core-9.0.39.jar and tomcat-embed-websocket-9.0.39.jar</notes>
+    <cve>CVE-2020-17527</cve>
```
Comment (timja) [inline config/owasp/suppressions.xml:46]: upgrade tomcat then like everyone else =/

File: src/main/resources/application-local.yaml (hunk 4 above)
Comment (amitchidrewar1301) [reply src/main/resources/application-local.yaml]: @timja Yeah this change was to sync up local prop with the actual application, but I have removed the above change as part of this pr

File: config/owasp/suppressions.xml [hunk 6]
```diff
@@ -42,7 +42,7 @@
   </suppress>
 
   <suppress>
-    <notes>This is transitive dependency coming for spring boot starter, CVE is for tomcat-embed-core-9.0.39.jar and tomcat-embed-websocket-9.0.39.jar</notes>
+    <notes>This is transitive dependency coming from spring boot starter, CVE is for tomcat-embed-core/Websocket jar. Latest version also have same CVE issues so expiry date cant be mentioned</notes>
```
Comment (timja) [inline config/owasp/suppressions.xml:45]: you should still put an expiry date, ~2 weeks should give time for it to be fixed

File: config/owasp/suppressions.xml (hunk 5 above)
Comment (amitchidrewar1301) [reply config/owasp/suppressions.xml:46]: @timja This is transitive dependency coming from spring boot starter, CVE is for tomcat-embed-core/Websocket jar. The latest version also has the same CVE issues. Don't have any other option apart from suppression.

File: build.gradle [hunk 7]
```diff
@@ -185,11 +176,11 @@ checkstyleTest.enabled(false)
...
       eachDependency { DependencyResolveDetails details ->
         if (details.requested.group in ['com.fasterxml.jackson.core', 'com.fasterxml.jackson.module', 'com.fasterxml.jackson.datatype']) {
-          details.useVersion '2.10.1'
+          details.useVersion '2.12.0'
         }
 
         if (details.requested.name == 'jackson-databind') {
-          details.useVersion '2.10.2'
+          details.useVersion '2.12.0'
         }
```
Comment (timja) [inline build.gradle:184]: shouldn't be needed as you have the group exclusion

File: config/owasp/suppressions.xml (hunk 5 above)
Comment (timja) [reply config/owasp/suppressions.xml:46]: this doesn't seem to be correct, if you read the advisory you'll see only up to 39 has that issue
https://nvd.nist.gov/vuln/detail/CVE-2020-17527

41 is patched, if you update to that you will pick up the fix,
https://mvnrepository.com/artifact/org.apache.tomcat.embed/tomcat-embed-websocket/9.0.41

**Review by timja — CHANGES_REQUESTED** [review]
dependency check exclusion is not right from what i can see

File: config/owasp/suppressions.xml (hunk 5 above)
Comment (amitchidrewar1301) [reply config/owasp/suppressions.xml]: @timja I have tried with the 10.0.0 version which had CVE present, but Yeah with 9.0.41 all sorted now. I have removed the suppression and upgraded tomcat versions.

File: config/owasp/suppressions.xml (hunk 6 above)
Comment (amitchidrewar1301) [reply config/owasp/suppressions.xml]: @timja It's no longer applicable as CVE suppression is removed now

File: build.gradle (hunk 7 above)
Comment (amitchidrewar1301) [reply build.gradle]: Yep made the necessary changes.

**Review by timja — APPROVED** [review]

**Review by YasharRahvar — APPROVED** [review]

**Review by krishnanuthalapati — APPROVED** [review]

**Pull request merged**